tests/unit/components/pages/donation_form/single_page_form/PaymentSection.spec.ts

```diff
@@ -0,0 +1,70 @@
+import { createStore } from 'vuex';
+import { mount, VueWrapper } from '@vue/test-utils';
+import PaymentSection from '@src/components/pages/donation_form/singlePageFormSections/PaymentSection.vue';
+import { Validity } from '@src/view_models/Validity';
+import Payment from '@src/store/payment';
+import PaymentSummary from '@src/components/pages/donation_form/PaymentSummary.vue';
+
+const createMockStore = ( paymentValidity: any, paymentState: any = {} ) => {
+	return createStore( {
+		state: {
+			payment: {
+				validity: paymentValidity,
+				values: paymentState,
+			},
+		},
+		getters: {
+			'payment/isExternalPayment': () => false,
+			'payment/isBankTransferPayment': () => false,
+			'payment/isDirectDebitPayment': () => paymentState.type === 'direct_debit',
+		},
+	});
+};
+
+const getWrapper = ( store: any ): VueWrapper<any> => {
+	return mount( PaymentSection, {
+		props: {
+			paymentAmounts: [ 10, 20, 30 ],
+			paymentIntervals: [ 1, 12 ],
+			paymentTypes: [ 'direct_debit', 'bank_transfer', 'credit_card' ],
+		},
+		global: {
+			plugins: [ store ],
+		},
+	} );
+};
+
+describe( 'PaymentSection.vue', () => {
+	it( 'should display the full form when the store is empty', () => {
+		const store = createMockStore(
+			{ type: Validity.INVALID, amount: Validity.INVALID },
+			{ amount: 0, interval: '', type: '' } );
+		const wrapper = getWrapper( store );
+
+		expect( wrapper.find( 'form[name="laika-donation-payment"]' ).exists() ).toBe( true );
+		// expect( wrapper.findComponent( Payment).exists() ).toBe( true );
+		// expect( wrapper.findComponent( PaymentSummary ).exists() ).toBe( false );
+	});
+
+	it( 'should display the summary when the payment data in the store is valid', () => {
+		const store = createMockStore(
```

Comment:
Make this test an async one and this can be:

```js
const store = createStore();
await store.dispatch( action( 'payment', 'setAmount' ), '1000' );
await store.dispatch( action( 'payment', 'setInterval' ), 'monthly' );
await store.dispatch( action( 'payment', 'setType' ), 'credit_card' );
```